### PR TITLE
[Forwardport] Fixed condition with usage isPost method

### PR DESCRIPTION
**Pull request by gelanivishal**
### Original Pull Request 
https://github.com/magento/magento2/pull/12626

- removed private function isPost
- removed unused imports
- added request method isPost
- added interface that expands HTTP request class

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @gelanivishal. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: lib/internal/Magento/Framework/App/HttpRequestInterface.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\App;
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/App/HttpRequestInterface.php]: Please define strict type declaration for this file

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
Hi @gelanivishal I fixed this issue myself. Let's wait when Travis will run all tests again

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-2431](https://jira.corp.magento.com/browse/ENGCOM-2431) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @gelanivishal. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.0. 
Please check the release notes for final confirmation.

**Pull request merged**